src/main/java/com/aws/greengrass/dependency/EZPlugins.java

```diff
@@ -275,6 +196,73 @@ public String[] list(boolean trusted) {
         return s.toArray(new String[0]);
     }
 
+    /**
+     * Delete all plugins.
+     *
+     * @return this
+     * @throws IOException if deletion fails
+     */
+    public EZPlugins clearCache() throws IOException {
+        IOException ioe = new IOException("One or more file deletion failed");
+        walk(cacheDirectory, p -> {
+            if (p.toString().endsWith(JAR_FILE_EXTENSION)) {
+                try {
+                    Files.delete(p);
+                } catch (IOException e) {
+                    ioe.addSuppressed(e);
+                }
+            }
+        });
+        if (ioe.getSuppressed().length > 0) {
+            throw ioe;
+        }
+        return this;
+    }
+
+    /**
+     * Load a jar from a URL into the plugin cache.
+     *
+     * @param trusted true if the plugin should be set as trusted
+     * @param u       URL to load the jar from
+     * @return this
+     * @throws IOException if loading fails
+     */
+    public EZPlugins loadToCache(boolean trusted, URL u) throws IOException {
+        String nm = Utils.namePart(u.getPath());
+        if (!nm.endsWith(JAR_FILE_EXTENSION)) {
+            throw new IOException("Only .jar files can be cached: " + u);
+        }
+        Path d = (trusted ? trustedCacheDirectory : untrustedCacheDirectory).resolve(nm);
+        Files.copy(u.openStream(), d, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *openStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *openStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).